packages/observable-client/src/chainHead/chainHead.ts

```diff
@@ -191,10 +207,10 @@ export const getChainHead$ = (chainHead: ChainHead) => {
 
     cache.set(hash, hashCache)
 
-    const connector = new ReplaySubject<T>()
+    let connector: ReplaySubject<T>
     const result = stream.pipe(
       share({
-        connector: () => connector,
+        connector: () => (connector = new ReplaySubject()),
```

Comment:
> Otherwise new calls to `upsertCachedStream` will get the stale ReplaySubject instead of the new one. Not sure if this is what we want though.

not really, b/c it's only set to the connector once it completes. So, I really think that this is the behaviour that we want.